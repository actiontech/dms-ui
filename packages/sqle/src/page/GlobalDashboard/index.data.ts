import { ProjectProjectPriorityEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import { t } from '../../locale/index';

export const ProjectPriorityDictionary: {
  [key in ProjectProjectPriorityEnum]: string;
} = {
  [ProjectProjectPriorityEnum.high]: t('globalDashboard.high'),
  [ProjectProjectPriorityEnum.medium]: t('globalDashboard.medium'),
  [ProjectProjectPriorityEnum.low]: t('globalDashboard.low')
};

export const ProjectPriorityOptions: Array<{
  label: string;
  value: ProjectProjectPriorityEnum;
}> = [
  {
    label: ProjectPriorityDictionary[ProjectProjectPriorityEnum.high],
    value: ProjectProjectPriorityEnum.high
  },
  {
    label: ProjectPriorityDictionary[ProjectProjectPriorityEnum.medium],
    value: ProjectProjectPriorityEnum.medium
  },
  {
    label: ProjectPriorityDictionary[ProjectProjectPriorityEnum.low],
    value: ProjectProjectPriorityEnum.low
  }
];
