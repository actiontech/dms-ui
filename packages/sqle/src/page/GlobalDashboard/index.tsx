import { PageHeader, SegmentedTabs } from '@actiontech/shared';
import useDashboardFilter from './hooks/useDashboardFilter';
import GlobalDashboardTableFilter from './components/TableFilter';
import { useState, useMemo } from 'react';
import { GlobalDashBoardSegmentedEnum } from './index.type';
import { SegmentedTabsProps } from '@actiontech/shared/lib/components/SegmentedTabs/index.type';
import { useTranslation } from 'react-i18next';
import { Space } from 'antd';
import PendingWorkOrder from './List/PendingWorkOrder';
import PendingSql from './List/PendingSql';
import InitiatedWorkOrder from './List/InitiatedWorkOrder';
import eventEmitter from '../../utils/EventEmitter';
import EmitterKey from '../../data/EmitterKey';
import { TableRefreshButton } from '@actiontech/shared/lib/components/ActiontechTable';

const GlobalDashBoard = () => {
  const { t } = useTranslation();

  const {
    projectOptions,
    instanceIDOptions,
    form,
    getInstanceListLoading,
    filterValues,
    updateFilterValue,
    refreshStatistics,
    pendingSqlStatistics,
    pendingWorkflowOrderStatistics,
    initiatedWorkflowOrderStatistics
  } = useDashboardFilter();

  const [activeKey, setActiveKey] = useState(
    GlobalDashBoardSegmentedEnum.PendingWorkOrder
  );

  const onRefresh = () => {
    if (activeKey === GlobalDashBoardSegmentedEnum.PendingWorkOrder) {
      eventEmitter.emit(EmitterKey.Refresh_Global_Dashboard_Pending_Work_Order);
    } else if (activeKey === GlobalDashBoardSegmentedEnum.PendingSqlRecord) {
      eventEmitter.emit(EmitterKey.Refresh_Global_Dashboard_Pending_Sql);
    } else if (activeKey === GlobalDashBoardSegmentedEnum.InitiatedWorkOrder) {
      eventEmitter.emit(
        EmitterKey.Refresh_Global_Dashboard_Initiated_Work_Order
      );
    }
    refreshStatistics();
  };

  const tabItems = useMemo<SegmentedTabsProps['items']>(() => {
    return [
      {
        label: (
          <Space>
            {t('globalDashboard.pendingWorkOrder')}
            {pendingWorkflowOrderStatistics}
          </Space>
        ),
        value: GlobalDashBoardSegmentedEnum.PendingWorkOrder,
        children: (
          <PendingWorkOrder
            filterValues={filterValues}
            updateFilterValue={updateFilterValue}
          />
        ),
        destroyInactivePane: true
      },
      {
        label: (
          <Space>
            {t('globalDashboard.pendingSql.title')}
            {pendingSqlStatistics}
          </Space>
        ),
        value: GlobalDashBoardSegmentedEnum.PendingSqlRecord,
        children: (
          <PendingSql
            filterValues={filterValues}
            updateFilterValue={updateFilterValue}
          />
        ),
        destroyInactivePane: true
      },
      {
        label: (
          <Space>
            {t('globalDashboard.initiatedWorkOrder')}
            {initiatedWorkflowOrderStatistics}
          </Space>
        ),
        value: GlobalDashBoardSegmentedEnum.InitiatedWorkOrder,
        children: (
          <InitiatedWorkOrder
            filterValues={filterValues}
            updateFilterValue={updateFilterValue}
          />
        ),
        destroyInactivePane: true
      }
    ];
  }, [
    t,
    filterValues,
    updateFilterValue,
    pendingSqlStatistics,
    pendingWorkflowOrderStatistics,
    initiatedWorkflowOrderStatistics
  ]);

  return (
    <>
      <PageHeader
        title={
          <Space>
            {t('globalDashboard.pageTitle')}
            <TableRefreshButton refresh={onRefresh} />
          </Space>
        }
      />
      <GlobalDashboardTableFilter
        form={form}
        projectOptions={projectOptions}
        instanceIDOptions={instanceIDOptions}
        getInstanceListLoading={getInstanceListLoading}
      />
      <SegmentedTabs
        activeKey={activeKey}
        onChange={(key) => {
          setActiveKey(key);
        }}
        items={tabItems}
      />
    </>
  );
};

export default GlobalDashBoard;
