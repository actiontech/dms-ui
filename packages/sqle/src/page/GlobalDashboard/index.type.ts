import { ListProjectV2ProjectPriorityEnum } from '@actiontech/shared/lib/api/base/service/common.enum';
import { FormInstance, SelectProps } from 'antd';

export type GlobalDashboardFilterType = {
  projectId?: string;
  instanceId?: string;
  projectPriority?: ListProjectV2ProjectPriorityEnum;
};

export type GlobalDashboardTableFilterProps = {
  form: FormInstance<GlobalDashboardFilterType>;
  projectOptions: SelectProps['options'];
  instanceIDOptions: SelectProps['options'];
  getInstanceListLoading: boolean;
};

export enum GlobalDashBoardSegmentedEnum {
  PendingWorkOrder = 'PendingWorkOrder',
  PendingSqlRecord = 'PendingSqlRecord',
  InitiatedWorkOrder = 'InitiatedWorkOrder'
}

export type GlobalDashboardListProps = {
  filterValues: GlobalDashboardFilterType;
  updateFilterValue: (
    key: keyof GlobalDashboardFilterType,
    value?: string
  ) => void;
};
