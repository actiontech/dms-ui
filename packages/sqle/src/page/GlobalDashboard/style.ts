import { styled } from '@mui/material/styles';
import { FilterContainerStyleWrapper } from '@actiontech/shared/lib/components/ActiontechTable/components/style';

export const GlobalDashboardFilterStyleWrapper = styled(
  FilterContainerStyleWrapper
)`
  & .ant-form.ant-form-horizontal {
    .ant-select-selector,
    .custom-search-input {
      font-size: 13px !important;
    }
  }

  & .custom-select-namespace {
    width: 280px;
  }
`;

export const PendingSqlTableStyleWrapper = styled('div')`
  .ant-tag {
    width: max-content;
  }
`;
