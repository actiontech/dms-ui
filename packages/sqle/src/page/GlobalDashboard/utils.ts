import queryString from 'query-string';
import { IGetGlobalWorkflowsV1Params } from '@actiontech/shared/lib/api/sqle/service/workflow/index.d';

export const paramsSerializer = (params: IGetGlobalWorkflowsV1Params) =>
  queryString.stringify(params, {
    arrayFormat: 'none'
  });
