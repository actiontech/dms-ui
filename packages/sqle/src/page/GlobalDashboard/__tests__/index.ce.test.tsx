/**
 * @test_version ce
 */
import { cleanup, act } from '@testing-library/react';
import { mockUseCurrentProject } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentProject';
import { mockUseCurrentUser } from '@actiontech/shared/lib/testUtil/mockHook/mockUseCurrentUser';
import { mockUseDbServiceDriver } from '@actiontech/shared/lib/testUtil/mockHook/mockUseDbServiceDriver';
import workflow from '../../../testUtils/mockApi/execWorkflow';
import sqlManage from '../../../testUtils/mockApi/sqlManage';
import instance from '../../../testUtils/mockApi/instance';
import { superRender } from '../../../testUtils/customRender';
import GlobalDashboard from '../index';

describe('sqle/GlobalDashboard ce', () => {
  let getGlobalWorkflowsSpy: jest.SpyInstance;
  let getGlobalSqlManageStatisticsSpy: jest.SpyInstance;
  let getGlobalWorkflowStatisticsSpy: jest.SpyInstance;
  beforeEach(() => {
    jest.useFakeTimers();
    mockUseCurrentProject();
    mockUseCurrentUser();
    mockUseDbServiceDriver();
    getGlobalWorkflowsSpy = workflow.getGlobalWorkflows();
    getGlobalWorkflowStatisticsSpy = workflow.getGlobalWorkflowStatistics();
    getGlobalSqlManageStatisticsSpy = sqlManage.getGlobalSqlManageStatistics();
  });

  afterEach(() => {
    jest.useRealTimers();
    cleanup();
  });

  it('render init snap shot', async () => {
    const { baseElement } = superRender(<GlobalDashboard />);
    await act(async () => jest.advanceTimersByTime(3000));
    expect(baseElement).toMatchSnapshot();
    expect(getGlobalWorkflowsSpy).toHaveBeenCalledTimes(1);
    expect(getGlobalSqlManageStatisticsSpy).not.toHaveBeenCalled();
    expect(getGlobalWorkflowStatisticsSpy).toHaveBeenCalledTimes(2);
  });
});
