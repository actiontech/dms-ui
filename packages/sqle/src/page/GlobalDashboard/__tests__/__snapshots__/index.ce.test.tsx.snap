// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqle/GlobalDashboard ce render init snap shot 1`] = `
<body>
  <div>
    <div
      class="actiontech-page-header-namespace css-1xspr0w"
    >
      <div
        class="title"
      >
        <div
          class="ant-space ant-space-horizontal ant-space-align-center"
        >
          <div
            class="ant-space-item"
            style="margin-right: 8px;"
          >
            Dashboard
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn ant-btn-default ant-btn-sm ant-btn-icon-only basic-button-wrapper css-geipcv"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <svg
                  class="custom-icon custom-icon-refresh"
                  height="14"
                  viewBox="0 0 14 14"
                  width="14"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.187 2.586a5.833 5.833 0 0 1 8.59 7.762L9.918 7h1.75a4.667 4.667 0 0 0-7.899-3.367zm7.626 8.828a5.833 5.833 0 0 1-8.59-7.762L4.083 7h-1.75a4.667 4.667 0 0 0 7.899 3.367z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        class="extra"
      />
    </div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center full-width-element css-1rtatmd"
    >
      <div
        class="ant-space-item"
      >
        <form
          class="ant-form ant-form-horizontal"
        >
          <div
            class="ant-space ant-space-horizontal ant-space-align-center"
          >
            <div
              class="ant-space-item"
              style="margin-right: 12px;"
            >
              <div
                class="ant-select ant-select-sm ant-select-borderless basic-select-wrapper custom-select-namespace css-pti5z8 ant-select-single ant-select-allow-clear ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="projectId_list_0"
                      aria-autocomplete="list"
                      aria-controls="projectId_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="projectId_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="projectId"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    <span
                      class="custom-select-placeholder css-qly0fu"
                    >
                      <span
                        class="custom-select-placeholder-prefix"
                      >
                        项目
                      </span>
                      <span
                        class="custom-select-placeholder-value"
                      >
                        请选择{{name}}
                      </span>
                    </span>
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                />
              </div>
            </div>
            <div
              class="ant-space-item"
              style="margin-right: 12px;"
            >
              <div
                class="ant-select ant-select-sm ant-select-borderless basic-select-wrapper custom-select-namespace css-pti5z8 ant-select-single ant-select-allow-clear ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="instanceId_list_0"
                      aria-autocomplete="list"
                      aria-controls="instanceId_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="instanceId_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="instanceId"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    <span
                      class="custom-select-placeholder css-qly0fu"
                    >
                      <span
                        class="custom-select-placeholder-prefix"
                      >
                        数据源
                      </span>
                      <span
                        class="custom-select-placeholder-value"
                      >
                        请选择{{name}}
                      </span>
                    </span>
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                />
              </div>
            </div>
            <div
              class="ant-space-item"
            >
              <div
                class="ant-select ant-select-sm ant-select-borderless basic-select-wrapper custom-select-namespace css-pti5z8 ant-select-single ant-select-allow-clear ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="projectPriority_list_0"
                      aria-autocomplete="list"
                      aria-controls="projectPriority_list"
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-owns="projectPriority_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="projectPriority"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    <span
                      class="custom-select-placeholder css-qly0fu"
                    >
                      <span
                        class="custom-select-placeholder-prefix"
                      >
                        项目优先级
                      </span>
                      <span
                        class="custom-select-placeholder-value"
                      >
                        请选择{{name}}
                      </span>
                    </span>
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                />
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="segmented-tabs-wrapper"
    >
      <div
        class="ant-row segmented-row-wrapper css-156vlto"
      >
        <div
          class="ant-segmented basic-segmented-wrapper css-192b3m1"
        >
          <div
            class="ant-segmented-group"
          >
            <label
              class="ant-segmented-item ant-segmented-item-selected"
            >
              <input
                checked=""
                class="ant-segmented-item-input"
                type="radio"
              />
              <div
                class="ant-segmented-item-label"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    待解决工单
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    10
                  </div>
                </div>
              </div>
            </label>
            <label
              class="ant-segmented-item"
            >
              <input
                class="ant-segmented-item-input"
                type="radio"
              />
              <div
                class="ant-segmented-item-label"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                  >
                    待解决SQL
                  </div>
                </div>
              </div>
            </label>
            <label
              class="ant-segmented-item"
            >
              <input
                class="ant-segmented-item-input"
                type="radio"
              />
              <div
                class="ant-segmented-item-label"
              >
                <div
                  class="ant-space ant-space-horizontal ant-space-align-center"
                >
                  <div
                    class="ant-space-item"
                    style="margin-right: 8px;"
                  >
                    发起的工单
                  </div>
                  <div
                    class="ant-space-item"
                  >
                    10
                  </div>
                </div>
              </div>
            </label>
          </div>
        </div>
      </div>
      <div
        class="lazy-load-wrapper segmented-item-content lazy-load-wrapper-show css-952l0m"
      >
        <div>
          <div
            class="ant-table-wrapper actiontech-table-namespace table-row-cursor css-fxd6ka"
          >
            <div
              class="ant-spin-nested-loading"
            >
              <div
                class="ant-spin-container"
              >
                <div
                  class="ant-table ant-table-scroll-horizontal"
                >
                  <div
                    class="ant-table-container"
                  >
                    <div
                      class="ant-table-content"
                      style="overflow-x: auto; overflow-y: hidden;"
                    >
                      <table
                        style="min-width: 100%; table-layout: auto;"
                      >
                        <colgroup />
                        <thead
                          class="ant-table-thead"
                        >
                          <tr>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              工单状态
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              工单名称
                            </th>
                            <th
                              class="ant-table-cell workflow-list-table-desc-column"
                              scope="col"
                            >
                              工单描述
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              创建时间
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              所属数据源
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              所属项目
                            </th>
                            <th
                              class="ant-table-cell"
                              scope="col"
                            >
                              项目优先级
                            </th>
                          </tr>
                        </thead>
                        <tbody
                          class="ant-table-tbody"
                        >
                          <tr
                            aria-hidden="true"
                            class="ant-table-measure-row"
                            style="height: 0px; font-size: 0px;"
                          >
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                            <td
                              style="padding: 0px; border: 0px; height: 0px;"
                            >
                              <div
                                style="height: 0px; overflow: hidden;"
                              >
                                 
                              </div>
                            </td>
                          </tr>
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="1846470650022072320"
                          >
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="css-3q8fl"
                              >
                                <svg
                                  fill="none"
                                  height="16"
                                  viewBox="0 0 16 16"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M6.875 1.516a2.25 2.25 0 0 1 2.25 0l3.928 2.267a2.25 2.25 0 0 1 1.125 1.95v4.535a2.25 2.25 0 0 1-1.125 1.948l-3.928 2.269a2.25 2.25 0 0 1-2.25 0l-3.928-2.268a2.25 2.25 0 0 1-1.125-1.95V5.733a2.25 2.25 0 0 1 1.125-1.949z"
                                    stroke="#7453DA"
                                    stroke-width="1.5"
                                  />
                                </svg>
                                <span>
                                  待审核
                                </span>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="css-10kos8e"
                              >
                                <svg
                                  height="14"
                                  viewBox="0 0 24 28"
                                  width="14"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M4 3.333v5.334h16V3.333h2.675c.732 0 1.325.594 1.325 1.325v21.35c0 .732-.593 1.325-1.325 1.325H1.325A1.325 1.325 0 0 1 0 26.01V4.658c0-.732.593-1.325 1.325-1.325zm4 17.334H5.333v2.666H8zm0-4H5.333v2.666H8zm0-4H5.333v2.666H8zm9.333-12V6H6.667V.667z"
                                    fill="#C3C6CD"
                                  />
                                </svg>
                                <a
                                  href="/sqle/project/700300/exec-workflow/1846470650022072320"
                                >
                                  dms-ui_20241016043751
                                </a>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell workflow-list-table-desc-column"
                            >
                              -
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              2024-10-16 16:38:03
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="ant-space ant-space-horizontal ant-space-align-center basic-tooltips-wrapper css-4jbtl9"
                              >
                                <div
                                  class="ant-space-item"
                                >
                                  <div
                                    class="ant-space ant-space-horizontal ant-space-align-center"
                                  >
                                    <div
                                      class="ant-space-item"
                                      style="margin-right: 8px;"
                                    >
                                      <a
                                        class="ant-typography"
                                      >
                                        mysql-1
                                      </a>
                                    </div>
                                    <div
                                      class="ant-space-item"
                                    >
                                      ...
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <a
                                class="ant-typography"
                              >
                                default
                              </a>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              高
                            </td>
                          </tr>
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="1846478104923475968"
                          >
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="css-3q8fl"
                              >
                                <svg
                                  fill="none"
                                  height="16"
                                  viewBox="0 0 16 16"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M6.875 1.516a2.25 2.25 0 0 1 2.25 0l3.928 2.267a2.25 2.25 0 0 1 1.125 1.95v4.535a2.25 2.25 0 0 1-1.125 1.948l-3.928 2.269a2.25 2.25 0 0 1-2.25 0l-3.928-2.268a2.25 2.25 0 0 1-1.125-1.95V5.733a2.25 2.25 0 0 1 1.125-1.949z"
                                    stroke="#7453DA"
                                    stroke-width="1.5"
                                  />
                                </svg>
                                <span>
                                  待审核
                                </span>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="css-10kos8e"
                              >
                                <svg
                                  height="14"
                                  viewBox="0 0 24 28"
                                  width="14"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M4 3.333v5.334h16V3.333h2.675c.732 0 1.325.594 1.325 1.325v21.35c0 .732-.593 1.325-1.325 1.325H1.325A1.325 1.325 0 0 1 0 26.01V4.658c0-.732.593-1.325 1.325-1.325zm4 17.334H5.333v2.666H8zm0-4H5.333v2.666H8zm0-4H5.333v2.666H8zm9.333-12V6H6.667V.667z"
                                    fill="#C3C6CD"
                                  />
                                </svg>
                                <a
                                  href="/sqle/project/700300/exec-workflow/1846478104923475968"
                                >
                                  mysql-1_20241016050733
                                </a>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell workflow-list-table-desc-column"
                            >
                              -
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              2024-10-16 17:07:40
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="ant-space ant-space-horizontal ant-space-align-center"
                              >
                                <div
                                  class="ant-space-item"
                                >
                                  <a
                                    class="ant-typography"
                                  >
                                    mysql-1
                                  </a>
                                </div>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <a
                                class="ant-typography"
                              >
                                default
                              </a>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              高
                            </td>
                          </tr>
                          <tr
                            class="ant-table-row ant-table-row-level-0"
                            data-row-key="1846474683185106944"
                          >
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="css-3q8fl"
                              >
                                <svg
                                  fill="none"
                                  height="16"
                                  viewBox="0 0 16 16"
                                  width="16"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M6.875 1.516a2.25 2.25 0 0 1 2.25 0l3.928 2.267a2.25 2.25 0 0 1 1.125 1.95v4.535a2.25 2.25 0 0 1-1.125 1.948l-3.928 2.269a2.25 2.25 0 0 1-2.25 0l-3.928-2.268a2.25 2.25 0 0 1-1.125-1.95V5.733a2.25 2.25 0 0 1 1.125-1.949z"
                                    stroke="#7453DA"
                                    stroke-width="1.5"
                                  />
                                </svg>
                                <span>
                                  待审核
                                </span>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <div
                                class="css-10kos8e"
                              >
                                <svg
                                  height="14"
                                  viewBox="0 0 24 28"
                                  width="14"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M4 3.333v5.334h16V3.333h2.675c.732 0 1.325.594 1.325 1.325v21.35c0 .732-.593 1.325-1.325 1.325H1.325A1.325 1.325 0 0 1 0 26.01V4.658c0-.732.593-1.325 1.325-1.325zm4 17.334H5.333v2.666H8zm0-4H5.333v2.666H8zm0-4H5.333v2.666H8zm9.333-12V6H6.667V.667z"
                                    fill="#C3C6CD"
                                  />
                                </svg>
                                <a
                                  href="/sqle/project/700300/exec-workflow/1846474683185106944"
                                >
                                  mysql-1_20241016045354
                                </a>
                              </div>
                            </td>
                            <td
                              class="ant-table-cell workflow-list-table-desc-column"
                            >
                              -
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              2024-10-16 16:54:05
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              -
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              <a
                                class="ant-typography"
                              >
                                default
                              </a>
                            </td>
                            <td
                              class="ant-table-cell"
                            >
                              高
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </div>
                </div>
                <ul
                  class="ant-pagination ant-table-pagination ant-table-pagination-right actiontech-table-pagination actiontech-table-pagination-fixed"
                >
                  <li
                    class="ant-pagination-total-text"
                  >
                    <span>
                      共 3 条数据
                    </span>
                  </li>
                  <li
                    aria-disabled="true"
                    class="ant-pagination-prev ant-pagination-disabled"
                    title="上一页"
                  >
                    <button
                      class="ant-pagination-item-link"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-label="left"
                        class="anticon anticon-left"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="left"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </li>
                  <li
                    class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
                    tabindex="0"
                    title="1"
                  >
                    <a
                      rel="nofollow"
                    >
                      1
                    </a>
                  </li>
                  <li
                    aria-disabled="true"
                    class="ant-pagination-next ant-pagination-disabled"
                    title="下一页"
                  >
                    <button
                      class="ant-pagination-item-link"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        aria-label="right"
                        class="anticon anticon-right"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="right"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                          />
                        </svg>
                      </span>
                    </button>
                  </li>
                  <li
                    class="ant-pagination-options"
                  >
                    <div
                      aria-label="页码"
                      class="ant-select ant-pagination-options-size-changer ant-select-single ant-select-show-arrow ant-select-show-search"
                    >
                      <div
                        class="ant-select-selector"
                      >
                        <span
                          class="ant-select-selection-search"
                        >
                          <input
                            aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                            aria-autocomplete="list"
                            aria-controls="rc_select_TEST_OR_SSR_list"
                            aria-expanded="false"
                            aria-haspopup="listbox"
                            aria-label="页码"
                            aria-owns="rc_select_TEST_OR_SSR_list"
                            autocomplete="off"
                            class="ant-select-selection-search-input"
                            id="rc_select_TEST_OR_SSR"
                            role="combobox"
                            type="search"
                            value=""
                          />
                        </span>
                        <span
                          class="ant-select-selection-item"
                          title="20 条/页"
                        >
                          20 条/页
                        </span>
                      </div>
                      <span
                        aria-hidden="true"
                        class="ant-select-arrow"
                        style="user-select: none;"
                        unselectable="on"
                      >
                        <span
                          aria-label="down"
                          class="anticon anticon-down ant-select-suffix"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="down"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                            />
                          </svg>
                        </span>
                      </span>
                    </div>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
